020-Connect/010-Postgres/001-direct-access.mdx

```diff
@@ -227,6 +227,23 @@ Tables created through the wire protocol will be missing the columns necessary t
 
 Please reference to the [special columns](/docs/concepts/data-model#special-columns) section to learn more about these columns.
 
+## Column metadata
+
+Xata uses SQL comments on columns to track some replication-related metadata about certain types of columns. These comments are only ever present on columns created using the Xata UI, and only on certain types of columns.
+
+The full list of column types and their associated metadata comments is as follows:
+
+| Xata Type | Metadata comment              | Notes                                 |
+| --------- | ----------------------------- | ------------------------------------- |
+| email     | `{"xata.type":"email"}`       |                                       |
+| vector    | `{"xata.search.dimension":n}` | dimension `n` is set in the UI        |
+| string    | `{"xata.type":"string"}`      |                                       |
+| text      | `{"xata.type":"text"}`        |                                       |
+| multiple  | `{"xata.type": "t"}`          | where `t` is the type of the multiple |
+| file      | `{"xata.file.dpa":false}`     |                                       |
```

Comment:
We also have comments for link type:
`{"xata.link": "referenced_table"}`
I didn't know we were passing the type for `multiple`, I don't think we use that in the backend at least. We do use the email/string/text though.  I thought the type`multiple` only supported strings (we map it internally to `text[]` only), so I wouldn't add it to the docs.

Vector has an additional setting, which is `hsnw`, although not sure if we want to advertise this one yet, since it's not really used and might affect search performance if the user decides to set it.
`{"xata.search.hnsw":false}`

Here's all the currently supported metadata:
https://github.com/xataio/xata/blob/main/xata/comment.go#L7-L13

Thank you so much for writing this up ❤️

Reply:
Updated to remove `multiple` and add `link`.

Reply:
FYI, I made a mistake in the UI creating the `multiple` column; there is no comment added to those comments and only strings are supported, as you said 😊